File: careerforge/src/main/java/com/careerforge/announcement/service/AnnouncementService.java
package com.careerforge.announcement.service;

import com.careerforge.announcement.dto.AnnouncementDto;
import com.careerforge.announcement.dto.CreateAnnouncementRequest;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.UUID;

public interface AnnouncementService {

    AnnouncementDto createAnnouncement(CreateAnnouncementRequest request, UUID adminId);

    Page<AnnouncementDto> getAnnouncements(Pageable pageable);

    void deleteAnnouncement(UUID id);
}

File: careerforge-frontend/src/services/announcementService.ts
import { apiClient } from '../config/api';
import { Announcement, CreateAnnouncementRequest } from '../types/announcement';

export const announcementService = {
  // Shared (student + admin)
  async getAnnouncements(page: number, size: number): Promise<any> {
    const response = await apiClient.get('/api/announcements', {
      params: { page, size },
    });
    return response.data.data;
  },

  // Admin only
  async createAnnouncement(data: CreateAnnouncementRequest): Promise<Announcement> {
    const response = await apiClient.post('/api/admin/announcements', data);
    return response.data.data;
  },

  async deleteAnnouncement(id: string): Promise<void> {
    await apiClient.delete(`/api/admin/announcements/${id}`);
  },
};
